src/System.Management.Automation/engine/Utils.cs

```diff
@@ -478,13 +478,13 @@ internal static string GetWindowsPowerShellVersionFromRegistry()
         }
 #endif
 
-        internal static string DefaultPowerShellAppBase => GetApplicationBase(DefaultPowerShellShellID);
+        internal static string DefaultPowerShellAppBase => GetApplicationBase();
 
-        internal static string GetApplicationBase(string shellId)
+        internal static string GetApplicationBase()
         {
-            // Use the location of SMA.dll as the application base.
+            // Use the location of SMA.dll as the application base or fallback to process home.
             Assembly assembly = typeof(PSObject).Assembly;
-            return Path.GetDirectoryName(assembly.Location);
+            return Path.GetDirectoryName(assembly.Location ?? AppContext.BaseDirectory ?? Environment.ProcessPath);
```

Comment:
This is not correct. Calling `Path.GetDirectoryName(AppContext.BaseDirectory)` will return the incorrect `PSHome` path.

I thought about a problem we have in application that hosting powershell -- when running `dotnet publish -r win7-x64` vs. running `dotnet publish`, the `S.M.A.dll` is placed in different locations -- and thus the modules bin-placed by `Microsoft.PowerShell.SDK` won't work for both scenarios. As of today, it works only for the `dotnet publish` scenario.

How about we don't use `S.M.A.dll` here, but change to:
```suggestion
            return AppContext.BaseDirectory ?? Path.GetDirectoryName(Environment.ProcessPath);
```
Then I will update the `Microsoft.PowerShell.SDK` package to always bin-place the Modules to the application root. It will make both `dotnet publish -r xxxx` and `dotnet publish` works.

Reply:
Yeah, we should probably stop relying on SMA.dll location.  I'll make your suggested change.

Reply:
It would be great to have tests (test projects) in SDK repo for host scenarios officially supported (managed, unmanaged, AOT hosts). They would be great docs and examples for developers too.